### ISSUE TITLE
Inconsistent accessibility checks  produce unexpected errors when types nested into interfaces are involved

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#38398]
```
        [Fact]
        public void InconsistentAccessibility()
        {
            var source1 =
@"
    interface I1
    {
        protected interface I2
        {
        }
    }

    class C1
    {
        protected interface I2
        {
        }
    }

    interface I3 : I1
    {
        protected I1.I2 M1();
    }

    class C3 : I1
    {
        protected void M1(I1.I2 x) { }
    }

    class C33 : C1
    {
        protected void M1(C1.I2 x) { }
    }
";
            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
                                                 parseOptions: TestOptions.Regular,
                                                 targetFramework: TargetFramework.NetStandardLatest);
            compilation1.VerifyDiagnostics(
                // (18,25): error CS0050: Inconsistent accessibility: return type 'I1.I2' is less accessible than method 'I3.M1()'
                //         protected I1.I2 M1();
                Diagnostic(ErrorCode.ERR_BadVisReturnType, "M1").WithArguments("I3.M1()", "I1.I2").WithLocation(18, 25),
                // (23,24): error CS0051: Inconsistent accessibility: parameter type 'I1.I2' is less accessible than method 'C3.M1(I1.I2)'
                //         protected void M1(I1.I2 x) { }
                Diagnostic(ErrorCode.ERR_BadVisParamType, "M1").WithArguments("C3.M1(I1.I2)", "I1.I2").WithLocation(23, 24)
                );
        }
```

Expected: No errors. Note, we don't produce the error for scenario when protected interface is nested into a class, the C33 case.
